master/internal/provisioner/aws_spot.go

```diff
@@ -0,0 +1,486 @@
+package provisioner
+
+import (
+	"encoding/base64"
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/ec2"
+	"github.com/google/uuid"
+	"github.com/pkg/errors"
+
+	"github.com/determined-ai/determined/master/pkg/actor"
+)
+
+// Spot instances are created asynchronously. You create a spot request, the
```

Comment:
Comment is accurate on how spot works, but innacurate for how the code is currently implemented (it describes a previous iteration)